WordPress/src/main/java/org/wordpress/android/ui/jetpack/backup/download/usecases/GetBackupDownloadStatusUseCase.kt

```diff
@@ -24,29 +24,34 @@ import javax.inject.Named
 
 const val DELAY_MILLIS = 1000L
 const val MAX_RETRY = 3
+const val DELAY_FACTOR = 2
 
 class GetBackupDownloadStatusUseCase @Inject constructor(
     private val networkUtilsWrapper: NetworkUtilsWrapper,
     private val activityLogStore: ActivityLogStore,
     @Named(BG_THREAD) private val bgDispatcher: CoroutineDispatcher
 ) {
+    private val tag = javaClass.simpleName
     suspend fun getBackupDownloadStatus(
         site: SiteModel,
         downloadId: Long? = null
     ) = flow {
-        val tag = javaClass.simpleName
         var retryAttempts = 0
+        var backoffDelay = DELAY_MILLIS
         while (true) {
             if (!isNetworkAvailable()) return@flow
 
             val result = activityLogStore.fetchBackupDownloadState(FetchBackupDownloadStatePayload(site))
             if (result.isError) {
-                if (retryAttempts++ >= MAX_RETRY) {
-                    AppLog.d(T.JETPACK_BACKUP, "$tag Exceeded 3 retries while fetching status")
-                    emit(RemoteRequestFailure)
-                    return@flow
+                when (exceedsRetryAttempts(retryAttempts++)) {
```

Comment:
🔍 Nitpick: We might want to use `if` instead of `when` here. Wdyt?

Reply:
I started with if and moved to when 🤦 , but will changed it back now that the delay is handled in the fun.

Reply:
Updated in 83916f7